src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -355,6 +364,118 @@ nlohmann::json SysInfo::getNetworks() const
     return networks;
 }
 
+std::pair<std::string, std::string> parseProcessInfo(char path[])
+{
+    // Open stat file
+    std::pair<std::string, std::string> processInfo;
+    std::ifstream file(path, std::ios::in | std::ios::binary);
+    std::string statContent;
+
+    // Get stat file content
+    std::getline(file, statContent);
+
+    // Parse PID and process name
+    int openParentesysPos = std::string(statContent).find("(");
+    int closeParentesysPos = std::string(statContent).find(")");
+    int spacePos = std::string(statContent).find(" ");
+
+    processInfo.first = std::string(statContent).substr(0, spacePos);
+    processInfo.second = std::string(statContent).substr(openParentesysPos + 1, closeParentesysPos - openParentesysPos - 1);
+
+    file.close();
+
+    return processInfo;
+}
+
+void findInodeMatch(char path[], std::vector<int64_t>& inodes, bool& matchInode, const nlohmann::json& JSONInput)
```

Comment:
This should be discussed with the team, if memory usage is relevant, with this approach it will increase significatively, also I don't think it will improve the performance. Finally, for gathering the information the recursion implementation will be needed since we have to look inside /proc/<id>/ and then parse the stat (or exe) file and the inner directories fd.

Reply:
Reimplemented using filesystem without recursion.